src/uniprotkb/adapters/ptmExchangeFeaturesConverter.ts

```diff
@@ -32,6 +35,24 @@ const convertPtmExchangePtms = (
       ])
     ),
   ];
+  const confidenceScores = Array.from(
+    new Set(
+      ptms.flatMap(({ dbReferences }) =>
+        dbReferences.map(({ properties }) => properties['Confidence score'])
+      )
+    )
+  );
+  let confidenceScore: ConfidenceScore | undefined;
+  if (!confidenceScores.length) {
+    logging.error('PTMeXchange PTM has no confidence score');
+  } else if (confidenceScores.length > 1) {
```

Comment:
```suggestion
  const confidenceScores = new Set(
    ptms.flatMap(({ dbReferences }) =>
      dbReferences.map(({ properties }) => properties['Confidence score'])
    )
  );
  let confidenceScore: ConfidenceScore | undefined;
  if (!confidenceScores.size) {
    logging.error('PTMeXchange PTM has no confidence score');
  } else if (confidenceScores.size > 1) {
```
Probably micro-optimisation, but I guess the extra array is not really needed

Reply:
Oh, that would maybe cause an issue with the stringification in the log.error for multiple values. I would say do an Array.from() there as it's not the branch of code that is expected to run the most

Reply:
I see what you mean, I did go back and forth on this a few times when I implemented it. We need to get the only element from `confidenceScores` (a few lines below) so for a set I would need to do `confidenceScores.values().next().value` which felt a bit strange (but valid). Overall, not sure which is preferable with regards to performance and readability?

Reply:
Actually didn't realise I can use destructuring assignment. I'll adapt to use a set then.